packages/attestation-service/src/models/attestation.ts

```diff
@@ -28,16 +46,51 @@ export default (sequelize: Sequelize) => {
     account: DataTypes.STRING,
     identifier: DataTypes.STRING,
     issuer: DataTypes.STRING,
+    countryCode: DataTypes.STRING,
+    phoneNumber: DataTypes.STRING,
+    message: DataTypes.STRING,
+    ongoingDeliveryId: DataTypes.STRING,
+    providers: DataTypes.STRING,
+    attempt: DataTypes.INTEGER,
     status: DataTypes.STRING,
-    smsProvider: DataTypes.STRING,
+    errors: DataTypes.STRING,
   }) as AttestationStatic
 
-  model.prototype.canSendSms = function() {
-    return [
-      AttestationStatus.DISPATCHING,
-      AttestationStatus.FAILED,
-      AttestationStatus.UNABLE_TO_SERVE,
-    ].includes(this.status)
+  model.prototype.key = function(): AttestationKey {
+    return { account: this.account, identifier: this.identifier, issuer: this.issuer }
+  }
+
+  model.prototype.provider = function(): string | null {
+    const pl = this.providers.split(',')
+    return this.providers ? pl[this.attempt % pl.length] : null
```

Comment:
what is this.attempt initialized as?  wondering if
1. this will ever be `null` or 
2. if on first attempt, this will actually set this to `1` in which case you wouldn't start with the 0 index provider

Reply:
Always zero: `sms/index.ts:183`